AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -166,8 +177,28 @@ - (void)enqueueItem:(id<MSLog>)item flags:(MSFlags)flags {
         [self checkPendingLogs];
       }
     }
+  };
+#if !TARGET_OS_OSX
+
+  // Process synchronously in case applicationWillTerminate was hit.
+  (self.applicationWillTerminateEntered ? dispatch_sync : dispatch_async)
+#else
+  dispatch_async
+#endif
+      (self.logsDispatchQueue, storeLogs);
+}
+
+#if !TARGET_OS_OSX
+- (void)applicationWillTerminate:(__unused UIApplication *)application {
+  self.applicationWillTerminateEntered = YES;
```

Comment:
After thinking - it's still not thread safe. There is no garantee that `dispatch_async` won't be scheduled after that